doc/api/esm.md

```diff
@@ -191,6 +191,12 @@ also be used within `"exports"` to define different package entry points per
 environment, including whether the package is referenced via `require` or via
 `import`.
 
+***Warning***: Introducing the exports field prevents consumers of a package
+for using any entry points that is not defined, including the `package.json`.
+This is a Semver-Major change. To make the introduction of `"exports"`
```

Comment:
Not everybody would agree that it's a semver-major change. For my packages, I do not consider deep imports to be part of the public API. Otherwise that would mean that internal refactoring (moving a file, or an internal function from one file to another for example) should also be considered semver-major.

Reply:
Yes, as it should.

The “exports” field changes the discussion from “how strictly do each of us interpret the semver spec” to “there’s no way you could consume something that i didn’t intend”. In the most liberal interpretation of semver, it might not be a breaking change, but its better to tell these people that it is, than to tell the rest of the spectrum that it’s not.

Reply:
I don't think that we should tell that it's not semver-major. But we can have this warning in the documentation (I'm in favor) without explicitly mentioning semverness. I would prefer something like: "This is a breaking change. To prevent the introduction of `"exports"` from affecting current consumers of a package, either ensure ..."

Reply:
What’s the benefit of not mentioning it? The worst case for your constituency is “an unnecessary major”, the worst case for mine is “breaking changes in a nonmajor” - i think we should be erring on the side of unnecessary majors here/

Reply:
```suggestion
This may be a Semver-Major change. To make the introduction of `"exports"`
```

I would encourage softening this a little bit, by at least allowing for the possibility that a package's API is something different from all-possible-entry-points. As far as I can tell, this would be the only place in the Node.js documentation where an opinion is espressed about the versioning of packages other than Node.js itself.

If the Node.js project does have a strong opinion on the versioning of JS packages, that opinion should probably be communicated more clearly and more strongly than in an aside like this. To be clear, I at least would almost certainly agree with it; I just find it surprising to encounter only in a place like this.

Reply:
The use of `Semver-Major` and `Semver-Minor` are no longer included in this doc change. I'm going to resolve this conversation PTAL